fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java

```diff
@@ -1359,13 +1359,30 @@ public ParseNode visitScalarSubquery(StarRocksParser.ScalarSubqueryContext conte
 
     @Override
     public ParseNode visitShowDatabasesStatement(StarRocksParser.ShowDatabasesStatementContext context) {
-        if (context.pattern != null) {
-            StringLiteral stringLiteral = (StringLiteral) visit(context.pattern);
-            return new ShowDbStmt(stringLiteral.getValue());
-        } else if (context.expression() != null) {
-            return new ShowDbStmt(null, (Expr) visit(context.expression()));
+        String catalog = null;
+        if (context.db != null) {
+            QualifiedName qualifiedName = getQualifiedName(context.db);
+            catalog = qualifiedName.toString();
+        }
+
+        if (catalog == null) {
+            if (context.pattern != null) {
+                StringLiteral stringLiteral = (StringLiteral) visit(context.pattern);
+                return new ShowDbStmt(stringLiteral.getValue());
+            } else if (context.expression() != null) {
+                return new ShowDbStmt(null, (Expr) visit(context.expression()));
+            } else {
+                return new ShowDbStmt(null, null, null);
+            }
         } else {
-            return new ShowDbStmt(null, null);
+            if (context.pattern != null) {
+                StringLiteral stringLiteral = (StringLiteral) visit(context.pattern);
+                return new ShowDbStmt(stringLiteral.getValue(), catalog);
+            } else if (context.expression() != null) {
+                return new ShowDbStmt(null, (Expr) visit(context.expression()), catalog);
+            } else {
+                return new ShowDbStmt(null, null, catalog);
+            }
```

Comment:
we do not need to check catalog isNull? just pass catalog, am i right? if it was true, we can simplify this code

Reply:
> 

done